### PR TITLE
People: Eslint fixes and use "Remove" user wording when appropriate

### LINKED ISSUE #8220: People: Adjust wording in the notice when removing a user from a site
#### Steps to reproduce
1. Remove a user from a site, using Calypso's people management.

#### What I expected

I expected a notice which stated that I had _removed_ the user successfully.

#### What happened instead

I received a notice stating that I had deleted the user, successfully. This is ambiguous, and may be scary to some users.

#### Screenshot / Video

image

### PR DESCRIPTION
**Pull request by ebinnion**
Fixes #8220

In #8220, @mattyza reported that using "delete" as a success notice when we are actually removing a user could be confusing to users. We hadn't previously distinguished between the two because both actions called a delete user endpoint, but on multisites, we actually remove instead of delete.

Because of this, if we're deleting a user on a multisite, I updated phrasing to user "remove".

I also made several ESLint fixes as I went through the code, which made the PR a bit bigger than it probably needed to be. 

To test:

- Checkout `update/adjust-remove-user-wording` branch
- Go to `/people/team/$site` where `$site` is a single site installation
- Edit a user and ensure there are no errors
- Delete the user and ensure you get the "delete" wording
- Do the same for a multisite install, but ensure you get the "remove" wording

cc @lezama for code review and @rickybanister for a design review

### CODE REVIEW

File: client/my-sites/people/edit-team-member-form/index.jsx [hunk 1]
```diff
@@ -50,23 +52,21 @@ var EditUserForm = React.createClass( {
...
 		);
 	},
 
 	getChangedSettings: function() {
-		var originalUser = this.getStateObject( this.props.user ),
-			changedKeys;
-
-		changedKeys = filter( this.getAllowedSettingsToChange(), function( setting ) {
+		const originalUser = this.getStateObject( this.props.user );
+		const changedKeys = filter( this.getAllowedSettingsToChange(), function( setting ) {
```
Comment (lezama) [inline client/my-sites/people/edit-team-member-form/index.jsx]: we could use an arrow function here and remove the `bind` call

File: client/my-sites/people/edit-team-member-form/index.jsx [hunk 2]
```diff
@@ -268,19 +267,19 @@ module.exports = React.createClass( {
 			return;
 		}
 
-		let removeUserSuccessful = PeopleLog.getCompleted( function( log ) {
+		const removeUserSuccessful = PeopleLog.getCompleted( function( log ) {
```
Comment (lezama) [inline client/my-sites/people/edit-team-member-form/index.jsx]: we could use an arrow function here and remove the `bind` call

File: client/my-sites/people/edit-team-member-form/index.jsx [hunk 3]
```diff
@@ -1,42 +1,44 @@
...
+import Card from 'components/card';
+import FormLabel from 'components/forms/form-label';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormTextInput from 'components/forms/form-text-input';
+import FormButton from 'components/forms/form-button';
+import FormButtonsBar from 'components/forms/form-buttons-bar';
+import PeopleProfile from 'my-sites/people/people-profile';
+import UsersStore from 'lib/users/store';
+import UsersActions from 'lib/users/actions';
+import user from 'lib/user';
```
Comment (lezama) [inline client/my-sites/people/edit-team-member-form/index.jsx]: we removed the `()` call to `require( 'lib/user' )`, I am getting a ` _user2.default.get is not a function` error when testing :)

Comment (ebinnion) [inline client/my-sites/people/edit-team-member-form/index.jsx]: I fixed this last night, although I may not have pushed it up, or maybe you just had a slightly outdated version. 😝 Will you test again?

**Comment by ebinnion** [comment]
@lezama - I have updated the PR after your feedback.

File: client/my-sites/people/edit-team-member-form/index.jsx (hunk 3 above)
Comment (lezama) [inline client/my-sites/people/edit-team-member-form/index.jsx]: I used the live branch!

**Comment by lezama** [comment]
Thanks! :shipit:

**Pull request merged**